### ISSUE TITLE
[pdcontrol] 'isvisible' when loading the patch doesn't work

### ISSUE THREAD

**Issue by porres** [pure-data/pure-data#926]
When loading a patch with a loadbang, it says it is invisible. But asking anytime later it says it is visible.

[isvisible.pd.zip](https://github.com/pure-data/pure-data/files/4441154/isvisible.pd.zip)

**Comment by umlaeute** [comment]
well, i would say, strictly speaking the current behaviour is correct:

the `[loadbang]` fires when the patch has been loaded, which is before it has been made visible (so the patch is invisible, and `[pdcontrol]` correctly reports "0").

i don't know what you are trying to achieve (the only thing that comes to my mind is guessing whether the user has opened a .pd-file as a top-level patch or as an abstraction), but the "isvisible" state is probably the wrong tool for that.

i'd like to close this as "wontfix"

**Comment by porres** [comment]
my intention was to check wether the canvas was in a GOP, which [pdcontrol] reports as visible, but only after I put some delay after the loadbang.

"isgraphed" can be another thing pdcontrol gives us instead, and it correctly reports it at loadbang.

I don't need "toplevel" report, and I'd like to know of a good use case for it.

**Comment by umlaeute** [comment]
so can we close this?

> I don't need "toplevel" report, and I'd like to know of a good use case for it.

just because i can think of something that one would want to do does not make it a good use case ;-)

**Comment by porres** [comment]
> the "isvisible" state is probably the wrong tool for that.

there could be a 'toplevel' thing for that, Purr Data has it, I just don't know what it is good for...

> so can we close this?

yeah, I don't see how to deal with this, and designing an external that tells me if it is graphed is what I'm doping now... so...
